platforms/jvm/plugins-java-base/src/integTest/groovy/org/gradle/api/plugins/jvm/internal/DefaultJvmFeatureTest.groovy

```diff
@@ -16,13 +16,111 @@
 
 package org.gradle.api.plugins.jvm.internal
 
+import org.gradle.api.attributes.Bundling
+import org.gradle.api.attributes.Category
+import org.gradle.api.attributes.DocsType
+import org.gradle.api.attributes.Usage
+import org.gradle.api.internal.tasks.JvmConstants
 import org.gradle.api.plugins.JavaBasePlugin
 import org.gradle.api.plugins.JavaPluginExtension
 import org.gradle.api.tasks.SourceSet
 import org.gradle.test.fixtures.AbstractProjectBuilderSpec
 
-// There aren't many tests for this class yet, but there will likely be more as we continue defining and refactoring the component/feature/target model
+/**
+ * Tests {@link DefaultJvmFeature}.
+ */
 class DefaultJvmFeatureTest extends AbstractProjectBuilderSpec {
+
+    def "can create a single feature"() {
+        given:
+        project.plugins.apply(JavaBasePlugin)
+        def ext = project.getExtensions().getByType(JavaPluginExtension.class)
+
+        when:
+        def feature = createFeature("main")
+
+        then:
+        feature.name == "main"
+        feature.sourceSet == ext.sourceSets.getByName(SourceSet.MAIN_SOURCE_SET_NAME)
+        feature.runtimeClasspathConfiguration == project.configurations.getByName(JvmConstants.RUNTIME_CLASSPATH_CONFIGURATION_NAME)
+        feature.compileClasspathConfiguration == project.configurations.getByName(JvmConstants.COMPILE_CLASSPATH_CONFIGURATION_NAME)
+        feature.runtimeElementsConfiguration == project.configurations.getByName(JvmConstants.RUNTIME_ELEMENTS_CONFIGURATION_NAME)
+        feature.apiElementsConfiguration == project.configurations.getByName(JvmConstants.API_ELEMENTS_CONFIGURATION_NAME)
+        feature.implementationConfiguration == project.configurations.getByName(JvmConstants.IMPLEMENTATION_CONFIGURATION_NAME)
+        feature.runtimeOnlyConfiguration == project.configurations.getByName(JvmConstants.RUNTIME_ONLY_CONFIGURATION_NAME)
+        feature.compileOnlyConfiguration == project.configurations.getByName(JvmConstants.COMPILE_ONLY_CONFIGURATION_NAME)
+        project.configurations.getByName(JvmConstants.ANNOTATION_PROCESSOR_CONFIGURATION_NAME)
```

Comment:
There are no methods on a feature to get the annotation processor configuration -- since none of our code needs to access it. 

We just test that the side effect of creating the configuration succeeded